app/scripts/metamask-controller.js

```diff
@@ -288,8 +290,10 @@ module.exports = class MetamaskController extends EventEmitter {
       CurrencyController: this.currencyRateController,
       ShapeshiftController: this.shapeshiftController,
       InfuraController: this.infuraController.store,
-      ProviderApprovalController: this.providerApprovalController.store,
       OnboardingController: this.onboardingController.store,
+      // ProviderApprovalController
+      ProviderApprovalController: this.providerApprovalController.store,
```

Comment:
See question above regarding definitely needing distinct store and memstore

Reply:
Answered above